modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/shibari.dm

```diff
@@ -0,0 +1,331 @@
+		return ..()
+	set_greyscale(greyscale_colors, /datum/greyscale_config/shibari_rope/high)
+	return ..()
+
+/obj/item/stack/shibari_rope/split_stack(mob/user, amount)
+	. = ..()
+	if(.)
+		var/obj/item/stack/F = .
+		F.set_greyscale(greyscale_colors)
+
+/obj/item/stack/shibari_rope/can_merge(obj/item/stack/check)
+	if(check.greyscale_colors == greyscale_colors)
+		return ..()
+	else
+		return FALSE
+
+/obj/item/stack/shibari_rope/Initialize(mapload, new_amount, merge, list/mat_override, mat_amt)
+	. = ..()
+	RegisterSignal(src, COMSIG_ITEM_ATTACK, .proc/handle_roping)
+	if(!greyscale_colors)
+		var/new_color = "#"
+		for(var/i in 1 to 3)
+			new_color += num2hex(rand(0, 255), 2)
+		set_greyscale(colors = list(new_color))
+
+/obj/item/stack/shibari_rope/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+
+/obj/item/stack/shibari_rope/proc/handle_roping(datum/source, mob/living/carbon/attacked, mob/living/user)
+	SIGNAL_HANDLER
+
+	if(get_dist(user, src) > 1)
+		return
+	if(ishuman(attacked))
+		if(attacked.client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy))
+			switch(user.zone_selected)
+				if(BODY_ZONE_L_LEG)
+					INVOKE_ASYNC(src, .proc/handle_leg_tying, attacked, user)
+					return COMPONENT_CANCEL_ATTACK_CHAIN
+				if(BODY_ZONE_R_LEG)
+					INVOKE_ASYNC(src, .proc/handle_leg_tying, attacked, user)
+					return COMPONENT_CANCEL_ATTACK_CHAIN
+				if(BODY_ZONE_PRECISE_GROIN)
+					INVOKE_ASYNC(src, .proc/handle_groin_tying, attacked, user)
+					return COMPONENT_CANCEL_ATTACK_CHAIN
+				if(BODY_ZONE_CHEST)
+					INVOKE_ASYNC(src, .proc/handle_chest_tying, attacked, user)
+					return COMPONENT_CANCEL_ATTACK_CHAIN
+				if(BODY_ZONE_L_ARM)
+					INVOKE_ASYNC(src, .proc/handle_arm_tying, attacked, user)
+					return COMPONENT_CANCEL_ATTACK_CHAIN
+				if(BODY_ZONE_R_ARM)
+					INVOKE_ASYNC(src, .proc/handle_arm_tying, attacked, user)
+					return COMPONENT_CANCEL_ATTACK_CHAIN
+		else
+			to_chat(user, span_danger("Looks like [attacked] doesn't want you to do that."))
+
+/obj/item/stack/shibari_rope/proc/handle_groin_tying(mob/living/carbon/human/them, mob/living/user)
+	if(!(them.w_uniform))
+		them.visible_message(span_warning("[user] starts tying [them]'s groin!"),\
+			span_userdanger("[user] starts tying your groin!"),\
+			span_hear("You hear ropes being tightened."))
+		if(do_after(user, HAS_TRAIT(user, TRAIT_RIGGER) ? 20 : 60))
+			var/obj/item/stack/shibari_rope/split_rope = null
+			var/slow = 0
+			if(them?.dna?.mutant_bodyparts["taur"])
+				var/datum/sprite_accessory/taur/S = GLOB.sprite_accessories["taur"][them.dna.species.mutant_bodyparts["taur"][MUTANT_INDEX_NAME]]
+				if(S.hide_legs)
+					split_rope = split_stack(null, 2)
+					slow = 4
+				else
+					split_rope = split_stack(null, 1)
+			else
+				split_rope = split_stack(null, 1)
+			if(split_rope)
+				shibari_groin = new(src)
+				shibari_groin.slowdown = slow
```

Comment:
once again, what's with the random slowdown for people who are taurs

Reply:
Not the random, taurs legs can’t be tied, their legs debuff applied through groin

Reply:
what do you mean by "can't be tied", are you telling me they're immune to bolas and legcuffs?

Reply:
No that means that tying legs of, for example, snake making no sense